### PR TITLE
hb-06 done

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/com/dev/cinema/model/Ticket.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "tickets")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne
```
Comment (vicire) [inline src/main/java/com/dev/cinema/model/Ticket.java]: You need to change the assignment. You can sell many tickets for one movie session

File: src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    //TicketDao. Hint: use this dao in the ShoppingCartService, method addSession
```
Comment (vicire) [inline src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java]: Remove this

**Review by vicire — CHANGES_REQUESTED** [review]

**Review by vicire — APPROVED** [review]

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -46,5 +51,26 @@ public static void main(String[] args) throws AuthenticationException {
...
+
+        Ticket ticket1 = new Ticket();
+        ticket1.setUser(bobLogin);
+        ticket1.setMovieSession(movieSession);
+        Ticket ticket2 = new Ticket();
+        ticket2.setUser(bobLogin);
+        ticket2.setMovieSession(movieSession);
+        List<Ticket> tickets = new ArrayList<>();
+        tickets.add(ticket1);
+        tickets.add(ticket2);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's remove this and use `shoppingCartService.addSession()` to add ticket to cart

File: src/main/java/com/dev/cinema/Main.java [hunk 4]
```diff
@@ -46,5 +51,26 @@ public static void main(String[] args) throws AuthenticationException {
...
+
+        ShoppingCartService shoppingCartService =
+                (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(bobLogin);
+        shoppingCart.setUser(bobLogin);
+        shoppingCart.setTickets(tickets);
+        System.out.println(shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's remove this as well

File: src/main/java/com/dev/cinema/Main.java [hunk 5]
```diff
@@ -46,5 +51,26 @@ public static void main(String[] args) throws AuthenticationException {
...
+        shoppingCartService.addSession(movieSession,bobLogin);
+        System.out.println(shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: to view the changes in code, call `getByUser()` method as well

File: src/main/java/com/dev/cinema/Main.java [hunk 6]
```diff
@@ -46,5 +51,26 @@ public static void main(String[] args) throws AuthenticationException {
...
+        shoppingCartService.clear(shoppingCart);
+        System.out.println(shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: same here

File: src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/ShoppingCartDaoImpl.java]:
```suggestion
        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

File: src/main/java/com/dev/cinema/model/Ticket.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "tickets")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/Ticket.java]:
```suggestion
@Table(name = "ticket")
```
let's use either singular or plural for all the tables

File: src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java]: redundant empty line

File: src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/ShoppingCartServiceImpl.java]: if there is no shopping cart for the user - we still save ticket to the db, is it ok?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**